crates/filtering/src/hysteresis.rs

```diff
@@ -1,29 +1,217 @@
-pub fn greater_than_with_hysteresis(
+use std::ops::Range;
+
+pub fn greater_than_with_absolute_hysteresis(
+    last_evaluation: bool,
+    value: f32,
+    hysteresis: Range<f32>,
+) -> bool {
+    if hysteresis.contains(&value) {
+        last_evaluation
+    } else {
+        value >= hysteresis.end
+    }
+}
+
+#[allow(clippy::if_same_then_else)]
+pub fn less_than_with_absolute_hysteresis(
+    last_evaluation: bool,
+    value: f32,
+    hysteresis: Range<f32>,
+) -> bool {
+    if hysteresis.contains(&value) {
+        last_evaluation
+    } else {
+        value <= hysteresis.start
+    }
+}
+
+pub fn greater_than_with_relative_hysteresis(
     last_evaluation: bool,
     value: f32,
     threshold: f32,
-    hysteresis: f32,
+    hysteresis: Range<f32>,
 ) -> bool {
-    value
-        > threshold
-            + if last_evaluation {
-                -hysteresis
-            } else {
-                hysteresis
-            }
+    greater_than_with_absolute_hysteresis(
```

Comment:
```suggestion
    assert(hysteresis.start <= 0.0);
    assert(hysteresis.end >= 0.0);

    greater_than_with_absolute_hysteresis(
```
What do you think?
Maybe I'm now extremely cautious... :D Feel free to resolve, if you think it is fine in the current state.

Reply:
Done